webrtc.html

```diff
@@ -2244,13 +2226,37 @@ <h4>
                                   </li>
                                   <li>
                                     <p>
-                                      For each of the codecs that <var>description</var> negotiates for sending, execute the following steps:
+                                      Clear the <var>transceiver</var>.{{RTCRtpTransceiver/[[Sender]]}}.{{RTCRtpSender/[[NegotiatedCodecs]]}} slot.
+                                      <br>
+                                      Then, for each of the codecs that <var>description</var> negotiates for sending, execute the following steps:
                                       <ol>
                                         <li>Locate the matching codec description in <var>transceiver</var>.{{RTCRtpTransceiver/[[Sender]]}}.{{RTCRtpSender/[[SendCodecs]]}}.</li>
                                         <li>If the matching codec description is not found, abort these steps.</li>
                                         <li>Set the "enabled" flag in the matching codec description to "true".</li>
+                                        <li>Add the codec description to the sender's {{RTCRtpSender/[[NegotiatedCodecs]]}} internal slot.</li>
```

Comment:
This puts tuples in [[NegotiatedCodecs]], which while expedient, seems a bit confusing since they'll always have "enabled" true. Is this on purpose?

Also since these are internal objects kept in lists in internal slots, these entries are added by reference, so if we ever introduce code that sets an enabled flag to false, it would affect both [[SendCodecs]] and [[NegotiatedCodecs]] potentially causing the codec entry to disappear from getParameters(). Is this desirable?

Even if desirable, this behavior seems a bit hard to grasp from this prose.

The other thing I noticed is that `payloadType` isn't filled in.

While [[[SendCodecs]]](https://w3c.github.io/webrtc-pc/#dfn-sendcodecs) is _"a list of [tuple](https://infra.spec.whatwg.org/#tuple)s, each containing an [RTCRtpCodecParameters](https://w3c.github.io/webrtc-pc/#dom-rtcrtpcodecparameters) dictionary and an "enabled" boolean"_ it's actually initialized to _"an [implementation-defined](https://infra.spec.whatwg.org/#implementation-defined) list of [RTCRtpCodecCapability](https://w3c.github.io/webrtc-pc/#dom-rtcrtpcodeccapability) dictionaries"_

Spot the different dictionary types above. RTCRtpCodecParameters != RTCRtpCodecCapability

The former contains payloadType, the latter doesn't. So it needs to be filled in here I think.

It might be good to use language shown in https://infra.spec.whatwg.org/#example-tuple to be more precise. This includes declaring the tuple where it's created with member names like `codec` and `enabled`, and "Let _entry_ be ..." etc.

This would let us be more precise and be clear whether we want to put the tuples in [[NegotiatedCodecs]] or simply the _entry_.codec.

Reply:
The intent of "negotiatedCodecs" is to reflect the currently negotiated codecs. The "enabled" flag has no purpose there, so it's not carried along.
Good point about using RTCRtpCodecCapability and filling in the PT field - this accentuates the need for it being a copy not a reference.

Reply:
Actually RTCRtpCodecCapability is a dictionary with no extra members over RTCRtpCodec. We shouldn't be using it.